dvc/repo/plots/__init__.py

```diff
@@ -33,3 +102,113 @@ def modify(self, path, props=None, unset=None):
 
         dvcfile = Dvcfile(self.repo, out.stage.path)
         dvcfile.dump(out.stage, update_pipeline=True)
+
+
+def _collect_plots(repo, targets=None, rev=None):
+    def _targets_to_outs(targets):
+        for t in targets:
+            try:
+                (out,) = repo.find_outs_by_path(t)
+                yield out
+            except OutputNotFoundError:
+                logger.warning(
+                    "File '{}' was not found at: '{}'. It will not be "
+                    "plotted.".format(t, rev)
+                )
+
+    if targets:
+        outs = _targets_to_outs(targets)
+    else:
+        outs = (out for stage in repo.stages for out in stage.outs if out.plot)
+
+    return {str(out): _plot_props(out) for out in outs}
+
+
+def _plot_props(out):
+    if not out.plot:
+        raise DvcException(
+            f"'{out}' is not a plot. Use `dvc plots modify` to change that."
```

Comment:
plots modify can change a file that is not a plot into a plot? Or maybe I don't understand this message.

Reply:
Yes, it can. It's a feature.

Reply:
OK thanks. What I meant is that I didn't understand what the sentence meant. Thinking about it now I think I get it: a regular output is being given to a plots command right? And you can make it into a plot with plots modify. So 2 things:

1. I recommend something like this for message for now: ``"'{out}' is found in stage '{stg}' as an output but not as a plot. You can make it into a plot with `plots modify ...` ``
2. Actually, what's the command to make an output into a plot with plots modify? I'd like to document this feature.
3. Why not just try to plot the output instead and throw a formatting error if it's not valid?

Thanks

Reply:
1. Too long, and not any useful information than the one provided already.
2. `dvc plots modify output-file-name-that-is-not-a-plot.file_ext`
You can provide any other flags on the same command, like `--template` too.

Reply:
2. Thanks

1. The existing string is hard to read and understand. Can we come up with a better suggestion if you mine isn't liked? "to change that" to change what?

Reply:
``The given output file is not a known plot. Use `dvc plots modify {out}` to turn it into one.``

p.s. why is it bad that a message is relatively long? (2 sentences)

Reply:
Also, can plots modify make ANY file into a plot, or only other outputs? What about existing metrics?
Does it validate plot file format?
Thanks

Reply:
> p.s. why is it bad that a message to the user is relatively long? (2 sentences)

The longer the message the smaller percentage of people will read it.

> Also, can plots modify make ANY file into a plot, or only other outputs?

Only outputs. In current dvc-file design only outputs may become metrics or plots. Metric might become a plot. Not sure whether it will stop being a metric.

> Does it validate plot file format?

No. `dvc run` doesn't validate it either. Strictly speaking we don't have such thing as validation here, we can check whether loading it results in an error though.

Reply:
> The longer the message the smaller percentage of people will read it.

Arguable. But if the text is hard to understand, it doesn't matter how many people read it. I guess I'll just sent a PR with a suggested text instead of arguing here.

Thanks for the other info. TBH it seems a little obscure/ arbitrary behavior but I'm not opposed to it. I just think it would be best to re-think it. Not really my realm so I'll just ping @shcheklein and @efiop here 🙂 

> dvc run doesn't validate it... we can check whether loading it results in an error

Please note causally I opened an issue about this (not related to this conversation originally): #4068